### PR TITLE
Use arrays for potential OpenSSL lib filenames

### PR DESCRIPTION
**Pull request by zorael**
This should fix SSL loading on macOS while allowing for easy additions on so-name bumps on any platform.

Big disclaimer: I don't know enough to know if it's correct, I just know it seems to work. Tested on Manjaro/Arch, Windows 11 and macOS Catalina, all x86_64.

Just close if you'd prefer to write (or have already written) a solution yourself.

### CODE REVIEW

**Comment by adamdruppe** [comment]
I'll take a closer look in a bit and prolly merge then fix up just my internet is kinda crappy right now.

**Comment by zorael** [comment]
Cheers, thanks

File: http2.d [hunk 1]
```diff
@@ -3252,33 +3252,60 @@ version(use_openssl) {
...
+				if(ossllib_handle !is null) break;
+			}
 		} else version(Windows) {
-			version(X86_64)
+			version(X86_64) {
 				ossllib_handle = LoadLibraryW("libssl-1_1-x64.dll"w.ptr);
-			if(ossllib_handle is null)
-				ossllib_handle = LoadLibraryW("libssl32.dll"w.ptr);
-			version(X86_64)
 				oeaylib_handle = LoadLibraryW("libcrypto-1_1-x64.dll"w.ptr);
```
[the remark this answers is not in the document]
Comment (zorael) [reply http2.d:3283]: `libssl32.dll` is on :3287, if that's the one you mean?

**Comment by adamdruppe** [comment]
On Wed, Apr 20, 2022 at 10:04:04AM -0700, JR wrote:
>  				oeaylib_handle = LoadLibraryW("libcrypto-1_1-x64.dll"w.ptr);
> 
> `libssl32.dll` is on :3287, if that's the one you mean?

no, the one with -x64 at the end

**Comment by zorael** [comment]
I'm not seeing it but doh, just shows why it got lost. Yeah, please fixup if so.

I left the Windows X86_64 branch untouched since there was only one option (the `-x64.dll`s). It's hard to tell from the diff though.

```d
		} else version(Windows) {
			version(X86_64) {
				ossllib_handle = LoadLibraryW("libssl-1_1-x64.dll"w.ptr);
				oeaylib_handle = LoadLibraryW("libcrypto-1_1-x64.dll"w.ptr);
			} else {
```

My second pull request ever, yay.

**Comment by adamdruppe** [comment]
there we go, most of it whitespace since i unindented it again but this looks good now
https://github.com/adamdruppe/arsd/commit/582f4287f8d4b22570f1716b4ce293dc85a56c37

thanks for all this!

**Pull request merged**